0-common/schema-ddl/project/BuildSettings.scala

```diff
@@ -1,73 +1,57 @@
+/**
+ * Copyright (c) 2014-2017 Snowplow Analytics Ltd. All rights reserved.
  *
  * This program is licensed to you under the Apache License Version 2.0,
- * and you may not use this file except in compliance with the
- * Apache License Version 2.0.
- * You may obtain a copy of the Apache License Version 2.0 at
- * http://www.apache.org/licenses/LICENSE-2.0.
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
  *
  * Unless required by applicable law or agreed to in writing,
- * software distributed under the Apache License Version 2.0 is distributed on
- * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- * express or implied.  See the Apache License Version 2.0 for the specific
- * language governing permissions and limitations there under.
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
  */
-import bintray.BintrayPlugin._
-import bintray.BintrayKeys._
+
+// SBT
 import sbt._
 import Keys._
 
 object BuildSettings {
 
-  // Basic settings for our app
-  lazy val basicSettings = Seq[Setting[_]](
-    organization          :=  "com.snowplowanalytics",
-    version               :=  "0.7.0",
-    description           :=  "Set of Abstract Syntax Trees for various DDL and Schema formats",
-    scalaVersion          :=  "2.12.4",
-    crossScalaVersions    :=  Seq("2.11.12", "2.12.4"),
-    scalacOptions         :=  Seq(
-      "-deprecation",
-      "-encoding", "UTF-8",
-      "-feature",
-      "-unchecked",
-      "-Ywarn-unused-import",
-      "-Ywarn-nullary-unit",
-      "-Xfatal-warnings",
-      "-Xlint",
-      "-language:higherKinds",
-      "-Ypartial-unification",
-      "-Xfuture"),
-    scalacOptions in (Compile, doc) ++= Seq(
-      "-no-link-warnings" // Suppresses problems with Scaladoc @throws links
-    ),
-    scalacOptions in Test :=  Seq("-Yrangepos")
-  )
-
-  // Publish settings
-  lazy val publishSettings = bintraySettings ++ Seq[Setting[_]](
-    licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0.html")),
-    bintrayOrganization := Some("snowplow"),
-    bintrayRepository := "snowplow-maven"
+  lazy val compilerOptions = Seq(
+    "-deprecation",
+    "-encoding", "UTF-8",
+    "-feature",
+    "-language:existentials",
+    "-language:higherKinds",
+    "-language:implicitConversions",
+    "-unchecked",
+    "-Yno-adapted-args",
+    "-Ywarn-dead-code",
+    "-Ywarn-numeric-widen",
+    "-Ywarn-unused-import",
+    "-Xfuture",
+    "-Xlint"
   )
 
-  // Maven Central publishing settings
-  lazy val mavenCentralExtras = Seq[Setting[_]](
```

Comment:
We cannot remove `sbt-bintray` and maven settings as we're publishing this library on Maven Central via Bintray/JCenter, so we won't be able to publish now.